core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -680,16 +688,17 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
     else if(brokerIdSet.size == 1) // pick broker.id from meta.properties
       brokerId = brokerIdSet.last
 
-    brokerId
+
+    (brokerId, offlineDirs)
   }
 
   private def checkpointBrokerId(brokerId: Int) {
     var logDirsWithoutMetaProps: List[String] = List()
 
-    for (logDir <- config.logDirs) {
+    for (logDir <- config.logDirs if logManager.liveLogDirs.contains(new File(logDir))) {
```

Comment:
Not exactly the same because config.logDirs is list of string whereas logManager.liveLogDirs is list of File. I have updated the code to avoid the if statement.